### DISCUSSION TITLE
Add support for more datatypes to Query Builder filters

### DISCUSSION THREAD

**Discussion by vadimp** [grafana/clickhouse-datasource#330] [Ideas]
Currently, QB filters support only simple data types (strings, numbers, etc...) with relevant comparison operators.
Having support for Map, Array and may be a few more will enrich the possibility of QB.

For example, a simple filter by map value with known key is is not possible (ResourceAttributes is a map). 
```
AND ( ResourceAttributes['k8s.namespace.name'] = 'my-namespace' )
```
In the below screenshot, ResourceAttributes field is treated as a standard data type.
![image](https://user-images.githubusercontent.com/657528/226872243-1bbb374b-559d-417e-80a0-e14c4ce86cd7.png)

Specifically for maps, I don't expect any key listing functionality. The key should be provided by the user explicitly.

Arrays filtering should follow the same idea. The ability to specify:
- Array entry
- Array range
- Whole array

With the relevant comparison operators including:
- Equals (full array equality)
- Contains
- Starts with
- Element at position equals to value
- ...

**Comment by SpencerTorres** [comment]
As of `v4.x` you can write in column names and filter values. We even have special support for the map type where a field is populated with map suggestions:

![image](https://github.com/grafana/clickhouse-datasource/assets/28887171/8e481eea-3278-4daf-9c01-d5e7ae329439)
